modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -1351,3 +1359,92 @@ func (p *ProjectPipelineService) ListUsedRefs(ctx context.Context, params deftyp
 	}
 	return resp.Ref, nil
 }
+
+type pipelineCategoryRule struct {
```

Comment:
I think `PipelineStatisticsByCategory` is more proper than `pipelineCategoryRule`